### DISCUSSION TITLE
src/app modified application path is not working with filament

### DISCUSSION THREAD

**Discussion by abd-wazzan** [filamentphp/filament#1009] [Help] [answered]
### Package

filament/filament

### Package Version

v2.7

### Laravel Version

8.75

### Livewire Version

2.8

### PHP Version

8.0.0

### Bug description

I am using DDD, so my application folder is in 'src/app' not in the default laravel folder 'app', and I have my Domain Folder in 
'src/Domain', I tried to install the package and run 'sail artisan make:filament-resource User', the package generate the resource successfully then I modified the namespace of my user model in UserResource.php
the package could not discover the resource automatically so I registered it in filament.php config file.
```
    'resources' => [
        'namespace' => 'App\\Filament\\Resources',
        'path' => app_path('Filament/Resources'),
        'register' => [
            'App\\Filament\\Resources\\UserResource'
        ],
    ],
```

the package discovered the resource and the user's section is on my sidenav, when I make any action to the user's section like checking checkbox exception is thrown with text
'Symfony\Component\Finder\Exception\DirectoryNotFoundException
The "src/app/Http/Livewire/" directory does not exist.'

**Comment by bezhanSalleh** [comment]
> ### Package
> filament/filament
> 
> ### Package Version
> v2.7
> 
> ### Laravel Version
> 8.75
> 
> ### Livewire Version
> 2.8
> 
> ### PHP Version
> 8.0.0
> 
> ### Bug description
> I am using DDD, so my application folder is in 'src/app' not in the default laravel folder 'app', and I have my Domain Folder in 'src/Domain', I tried to install the package and run 'sail artisan make:filament-resource User', the package generate the resource successfully then I modified the namespace of my user model in UserResource.php the package could not discover the resource automatically so I registered it in filament.php config file.
> 
> ```
>     'resources' => [
>         'namespace' => 'App\\Filament\\Resources',
>         'path' => app_path('Filament/Resources'),
>         'register' => [
>             'App\\Filament\\Resources\\UserResource'
>         ],
>     ],
> ```
> 
> the package discovered the resource and the user's section is on my sidenav, when I make any action to the user's section like checking checkbox exception is thrown with text 'Symfony\Component\Finder\Exception\DirectoryNotFoundException The "src/app/Http/Livewire/" directory does not exist.'
> 
> ### Steps to reproduce
> _No response_
> 
> ### Relevant log output
> _No response_

all you need to resolve this issue is create a `Livewire` foler inside `src/app/Http/` which tbh it kinda suggests it 🍻

@abd-wazzan this also happens when you run `filament:upgrade` if you don't have that `Livewire` folder inside `app\Http\` but creating the folder resolves the issue. 

should be closed now 🖖

**Comment by abd-wazzan** [comment, marked as answer]
I tried to create the "livewire" folder manually and also created some livewire components to make sure the folder is ok, but it did not work.

**Reply by luilliarcec** [reply]
Greetings, I had the same problem. Change my project structure to 

```
- src
  - \App
  - \Domain
  - \Support
```

When you make those changes, you must modify the app.php inside the bootstrap directory. Maybe you have done it but you must have made the same mistake as me, however I will pass you part of the code of how my app.php looks.

```php

// Correct

$app = (new Support\Application(
    $_ENV['APP_BASE_PATH'] ?? dirname(__DIR__)
))->useAppPath(realpath(__DIR__.'/../src/App'));

// Incorrect

$app = (new Support\Application(
    $_ENV['APP_BASE_PATH'] ?? dirname(__DIR__)
))->useAppPath('src/App');

```

With that change, it already recognizes the filament resources and the livewire folder.

**Reply by abd-wazzan** [reply]
Thank you very much it did work.

**Comment by danharrin** [comment]
Hey, I'm gonna move this to a discussion as it's not really a bug in Filament, we don't officially support this DDD setup but I'm happy to assist you to get it working.

Make sure that you also change the namespace of the pages inside your resource's `Pages` directory.

Are other Livewire components working fine in your directory?
